insights/client/client.py

```diff
@@ -352,6 +332,28 @@ def _legacy_upload(config, pconn, tar_file, content_type, collection_duration=No
                 logger.error("All attempts to upload have failed!")
                 logger.error("Please see %s for additional information", config.logging_file)
                 raise RuntimeError('Upload failed.')
+
+    api_response = json.loads(upload.text)
```

Comment:
Assuming `upload` is a `requests.Response` object, you can simply call `upload.json()` instead.